src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/ODataDeltaFeedSerializer.cs

```diff
@@ -531,22 +567,42 @@ public async Task WriteDeltaLinkAsync(object graph, ODataWriter writer, ODataSer
             }
         }
 
-        private ODataDeletedResource GetDeletedResource(object graph)
+
+        private ODataDeletedResource GetDeletedResource(object graph, ResourceContext resourceContext, ODataResourceSerializer serializer, SelectExpandNode selectExpandNode, bool isUntyped)
         {
-            EdmDeltaDeletedEntityObject edmDeltaDeletedEntity = graph as EdmDeltaDeletedEntityObject;
-            if (edmDeltaDeletedEntity == null)
+            string navigationSource;
+            ODataDeletedResource deletedResource = serializer.CreateDeletedResource(selectExpandNode, resourceContext);
+
+            if (isUntyped)
             {
-                throw new SerializationException(Error.Format(SRResources.CannotWriteType, GetType().Name, graph.GetType().FullName));
+                EdmDeltaDeletedEntityObject edmDeltaDeletedEntity = graph as EdmDeltaDeletedEntityObject;
+                if (edmDeltaDeletedEntity == null)
+                {
+                    throw new SerializationException(Error.Format(SRResources.CannotWriteType, GetType().Name, graph.GetType().FullName));
+                }
+
+                deletedResource.Id = StringToUri(edmDeltaDeletedEntity.Id ?? string.Empty);
```

Comment:
Please confirm that passing an empty string to `StringToUri` won't result into an exception

Reply:
Have confirmed. It doesn't throw an exception.